core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -135,16 +142,41 @@ public long requestCount() {
     }
 
     this.opts = opts;
-    if (opts.indexCacheSize > 0) {
-      this.indexCache = new LruBlockCache(opts.indexCacheSize, CACHE_BLOCK_SIZE);
-    } else {
-      this.indexCache = new NoopCache();
-    }
 
-    if (opts.dataCacheSize > 0) {
-      this.dataCache = new LruBlockCache(opts.dataCacheSize, CACHE_BLOCK_SIZE);
-    } else {
-      this.dataCache = new NoopCache();
+    if (opts.indexCacheSize > 0 || opts.dataCacheSize > 0) {
+      ConfigurationCopy cc = new ConfigurationCopy();
+      cc.set(Property.TSERV_CACHE_FACTORY_IMPL, LruBlockCacheFactory.class.getName());
+      try {
+        factory = BlockCacheFactory.getInstance(cc);
```

Comment:
This will use Accumulo start to load classes.  Currently, Accumulo code that loads classes client side avoids using Accumulo start.